src/bot/music.rs

```diff
@@ -171,40 +184,69 @@ impl MusicBot {
         Ok(())
     }
 
-    pub async fn add_audio(&mut self, url: String, user: String) -> anyhow::Result<()> {
-        match youtube_dl::get_audio_download_from_url(url, &self.logger).await {
-            Ok(mut metadata) => {
-                metadata.added_by = user;
-                info!(self.logger, "Found source"; "url" => &metadata.url);
+    pub async fn add_audio(&mut self, uri: String, user: String) -> anyhow::Result<()> {
+        let metadata = if uri.starts_with("file://") && self.music_root.is_some() {
+            let path = self.music_root.as_ref().unwrap().join(&uri[7..]);
+            let path = match path.canonicalize() {
+                Ok(p) => p,
+                Err(e) => {
+                    self.send_message(format!("Invalid path: {}", e)).await?;
+                    return Err(e)?;
```

Comment:
maybe send message from the caller of the method using the error Display instead of duplicating?

Reply:
Meh, okay.